### ISSUE TITLE
ReactPDF.render is not a function

### ISSUE THREAD

**Issue by luskin** [diegomura/react-pdf#341]
**OS:** Mac OS X High Sierra

**React-pdf version:** ^1.0.0-alpha.18

**Description:** ReactPDF.render does not work and crashes the application

index.js
```
import React from 'react'
import Test from 'templates/Test'
import ReactPDF from '@react-pdf/renderer'

ReactPDF.render(<Test />, './example.pdf')
```

templates/Test/index.js (copied from README):
```
import React from 'react'
import { Document, Page, Text, View, StyleSheet } from '@react-pdf/renderer'

// Create styles
const styles = StyleSheet.create({
  page: {
    flexDirection: 'row',
    backgroundColor: '#E4E4E4',
  },
  section: {
    margin: 10,
    padding: 10,
    flexGrow: 1,
  },
})

// Create Document Component
const MyDocument = () => (
  <Document>
    <Page size="A4" style={styles.page}>
      <View style={styles.section}>
        <Text>Section #1</Text>
      </View>
      <View style={styles.section}>
        <Text>Section #2</Text>
      </View>
    </Page>
  </Document>
)
export default MyDocument
```

ERROR WHEN RUNNING:
```
TypeError: _react_pdf_renderer__WEBPACK_IMPORTED_MODULE_4__.default.render is not a function
Module../src/index.js
src/index.js:9
>  9 | ReactPDF.render(<Test />, './example.pdf')
```

**Comment by suenwl** [comment]
I'm still experiencing this issue. Any way to get around this?

**Comment by fang0rnz** [comment]
same problem here @suenwl

**Comment by fang0rnz** [comment]
Are you using electron? I found out that you need to import the renderer in node.js context instead of DOM context. I changed all my imports to window.require and now it works fine.

**Comment by ewolfe** [comment]
Same problem, not using electron. Literally just copy/pasted from the README as this is my first time using this project.

**Comment by diegomura** [comment]
Just bare in mind that `ReactPDF.render()` is a Node-only API to save pdfs in disk. If you are using this lib in a web environment (such as electron) this method won't be available and that's probably what's causing these issues

**Comment by ewolfe** [comment]
Got it. Might be worth making a note of that in the README. I was assuming the browser would've created a PDF file (via a blob) and immediately downloaded it.

**Comment by xwilluhn** [comment]
agree. should have a note in README saying ReactPDF.render() is a Node-only API

**Comment by xwilluhn** [comment]
Thanks. Do you have any suggestion what should I use for pure frontend PDF
render?
Xia

**Comment by AirborneEagle** [comment]
I am simply trying to generate a pdf on a button click in the UI. I am getting this exact same error.

**Comment by goodspeed666** [comment]
Hi @diegomura ,

Could you tell me how to save pdf file under `Web` mode ?
I use `ReactPDF.render` got the same error.
And I cannot recognize the `Web` or `Node` mode.
Is the `Node` mode only for command line ?

Thank you.

**Comment by diegomura** [comment]
Hi. There isn’t a node or web mode. It’s just about in which envorinment are you using this lib (web or “server”). 

To download documents in the browser please check PDFDownloadLink on the docs

**Comment by goodspeed666** [comment]
Hi @diegomura ,

Thank you for your detailed response.
I will try it 👍

**Comment by goodspeed666** [comment]
Hi @diegomura ,

It's done for the web.
May I know react-pdf could be use with the cordova ?

Thank you.

**Comment by anisg** [comment]
If like me you just want to have a simple function to download a pdf Document through the web, you can do:
```jsx
import { pdf } from '@react-pdf/renderer';

const saveBlob = (blob, filename) => {
  var a = document.createElement("a");
  document.body.appendChild(a);
  console.log(blob)
  a.style.display = "none";
      let url = window.URL.createObjectURL(blob);
      a.href = url;
      a.download = filename;
      a.click();
      window.URL.revokeObjectURL(url);
}

export const savePdf = async (document, filename) => {
  saveBlob(await pdf(document).toBlob(), filename);
};
```
then in your React code:
```jsx
savePdf(<MyDocument/>, "my-document.pdf")
````
note that the function `savePdf` is asynchrone (you may need to use `await`)

**Comment by mateo2181** [comment]
@anisg what version are you using? I've tried this but I get error Type mismatch.

**Comment by anisg** [comment]
@mateo2181 ,  my version of "@react-pdf/renderer" is "1.6.8"

**Comment by cybermaq** [comment]
ASI DE FACIL LO RESOLVI, MUCHOS NO RESPONDIERON NI DIERON SOLUCION, PERO AQUI LE MANDO LA SOLUCION:

`import React from 'react';
import ReactPDF, {
  Document,
  Page,
  Text,
  View,
  Image,
  StyleSheet,
  Font,
  PDFDownloadLink,
} from '@react-pdf/renderer';
import { Button } from '@material-ui/core/';

const styles = StyleSheet.create({
  page: {
    flexDirection: 'row',
    backgroundColor: '#E4E4E4',
  },
  section: {
    margin: 10,
    padding: 10,
    flexGrow: 1,
  },
})

export default function App() {

  const MyDoc = () => (
    <Document>
      <Page size="A4" style={styles.page}>
      <View style={styles.section}>
        <Text>Section #1</Text>
      </View>
      <View style={styles.section}>
        <Text>Section #2</Text>
      </View>
    </Page>
    </Document>
  )
  
  const Appp = () => (
    <div>
      <PDFDownloadLink document={<MyDoc />} fileName="somename.pdf">
        {({ blob, url, loading, error }) => (loading ? 'Loading document...' : 'Download now!')}
      </PDFDownloadLink>
    </div>
  )

  return (
    <div>
      <Button onClick={()=>Appp()}>
PDF expotar
      </Button>

      <Appp />
     </div>
  );
}

`

**Comment by bouncingbumble** [comment]
The [Download Link section](https://react-pdf.org/advanced#on-the-fly-rendering) in the docs covers the use case desired 

```
import { PDFDownloadLink, Document, Page } from '@react-pdf/renderer';

const MyDoc = () => (
  <Document>
    <Page>
      // My document data
    </Page>
  </Document>
);

const App = () => (
  <div>
    <PDFDownloadLink document={<MyDoc />} fileName="somename.pdf">
      {({ blob, url, loading, error }) =>
        loading ? 'Loading document...' : 'Download now!'
      }
    </PDFDownloadLink>
  </div>
);
```
